gulpfile.js

```diff
@@ -1,88 +1,183 @@
-var gulp = require('gulp');
-var bulk = require('gulp-sass-glob');
-var sass = require('gulp-sass');
-var postcss = require('gulp-postcss');
-var postcssPrefix = require('postcss-prefix-selector');
-var postcssDiscardDuplicates = require('postcss-discard-duplicates');
-var stripCssComments = require('gulp-strip-css-comments');
-var transformSelectors = require('gulp-transform-selectors');
-var civicrmScssRoot = require('civicrm-scssroot')();
-
-var bootstrapNamespace = '#bootstrap-theme';
-var outsideNamespaceRegExp = /^\.___outside-namespace/;
-
-gulp.task('sass:sync', civicrmScssRoot.update);
-
-gulp.task('sass:bootstrap', gulp.series('sass:sync', function buildBootstrapCSS () {
-  return gulp.src('scss/bootstrap/bootstrap.scss')
-    .pipe(bulk())
-    .pipe(sass({
-      outputStyle: 'compressed',
-      includePaths: civicrmScssRoot.getPath(),
-      precision: 10
-    }).on('error', sass.logError))
-    .pipe(stripCssComments({ preserve: false }))
-    .pipe(postcss([postcssPrefix({
-      prefix: bootstrapNamespace + ' ',
-      exclude: [/^html/, /^body/, /\.popover/, /\.tooltip/, /\.ta-hidden-input/, outsideNamespaceRegExp]
-    })]))
-    .pipe(transformSelectors(namespaceRootElements, { splitOnCommas: true }))
-    .pipe(transformSelectors(removeOutsideNamespaceMarker, { splitOnCommas: true }))
-    .pipe(gulp.dest('css/'));
-}));
-
-gulp.task('sass:civicrm', gulp.series('sass:sync', function buildCiviCRMCSS () {
-  return gulp.src('scss/civicrm/custom-civicrm.scss')
-    .pipe(bulk())
-    .pipe(sass({
-      outputStyle: 'compressed',
-      includePaths: civicrmScssRoot.getPath(),
-      precision: 10
-    }).on('error', sass.logError))
-    .pipe(stripCssComments({ preserve: false }))
-    .pipe(postcss([postcssPrefix({
-      prefix: '.crm-container ',
-      exclude: [/^body/, /page-civicrm/, /crm-container/,
-        /ui-datepicker/, /civicrm-menu/, /#root-menu-div/, /jstree-contextmenu/,
-        outsideNamespaceRegExp]
-    }), postcssDiscardDuplicates]))
-    .pipe(transformSelectors(removeOutsideNamespaceMarker, { splitOnCommas: true }))
-    .pipe(gulp.dest('css/'));
-}));
-
-gulp.task('sass', gulp.parallel('sass:bootstrap', 'sass:civicrm'));
-
-gulp.task('watch', function () {
-  gulp.watch(civicrmScssRoot.getWatchList(), gulp.parallel('sass'));
-});
+const gulp = require('gulp');
+const argv = require('yargs').argv;
+const PluginError = require('plugin-error');
 
-gulp.task('default', gulp.parallel('sass'));
+// sass
+{
```

Comment:
Why do you need a separate scope here?

Reply:
same reason you would use a IIFE, to not pollute the outer scope with vars and functions that is not necessary to have in there